src/hex/grid/vertex.rs

```diff
@@ -0,0 +1,142 @@
+use std::ops::Neg;
+
+use crate::{DiagonalDirection, Hex};
+
+use super::GridEdge;
+
+/// Hexagonal grid orientated vertex representation.
+#[derive(Debug, Clone, Copy)]
+pub struct GridVertex {
+    /// The coordinate of the edge
+    pub origin: Hex,
+    /// The direction the vertex points towards
+    pub direction: DiagonalDirection,
+}
+
+impl PartialEq for GridVertex {
```

Comment:
Same remarks as for GridEdge, a `fn is_same_vertexself, other: &Self) -> bool {` would be safer IMHO